### PR TITLE
fix(oauth): handle short-lived access tokens for profile server requests

### LINKED ISSUE #2785: Handle short-lived access tokens for fetching profile info
We currently cache the `accessToken` and don't attempt to refresh it if auth fails... we should fix that.

### PR DESCRIPTION
**Pull request by zaach**
Fixes #2785.

@vladikoff r?

### CODE REVIEW

**Comment by vladikoff** [comment]
```
app/tests/spec/models/account.js
  447:12  error  Expected indentation of 12 characters  indent
  448:14  error  Expected indentation of 14 characters  indent
  449:12  error  Expected indentation of 12 characters  indent
  450:14  error  Expected indentation of 14 characters  indent
  451:12  error  Expected indentation of 12 characters  indent
```

**Comment by vladikoff** [comment]
@zaach something happened to your git config, "From: mozilla <mozilla@MacBook-Pro.local>"

File: app/scripts/models/account.js [hunk 1]
```diff
@@ -90,18 +90,23 @@ define([
...
+          return self._fetchProfileOAuthToken();
         }
       });
 
       return promise;
     },
 
+    _fetchProfileOAuthToken: function () {
+      var self = this;
+      return self.createOAuthToken('profile:write')
```
Comment (vladikoff) [inline app/scripts/models/account.js]: nit: make constant? It was not before but if we are refactoring this a bit, might be a good idea.

**Comment by zaach** [comment]
@vladikoff I'm using a loner machine :sob:

**Pull request merged**